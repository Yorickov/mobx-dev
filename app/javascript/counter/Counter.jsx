import React from 'react';
import { observer } from 'mobx-react';

const Counter = observer((props) => {
  const { age, nickname, decrease, increase } = props.store;

  return (
    <div>
      <h1>{nickname}</h1>
      <h1>{age}</h1>
      <button data-test="decrease" onClick={decrease}>decrement</button>
      <button data-test="increase" onClick={increase}>increase</button>
    </div>
  );
}, { name: 'counter' });

export default Counter;
