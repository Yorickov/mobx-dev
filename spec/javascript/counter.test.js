import React from 'react';
import Counter from '../../app/javascript/components/Counter';
import counterStore from '../../app/javascript/stores/counterStore';

it('Counter', () => {
  const wrapper = mount(<Counter store={counterStore} />);
  expect(wrapper.render()).toMatchSnapshot();

  const dec = wrapper.find('button[data-test="decrease"]');
  const inc = wrapper.find('button[data-test="increase"]');

  dec.simulate('click');
  inc.simulate('click');
  inc.simulate('click');

  expect(wrapper.render()).toMatchSnapshot();
});
