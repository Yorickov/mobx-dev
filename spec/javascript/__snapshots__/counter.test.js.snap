// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter 1`] = `
<div>
  <h1>
    Dan_23
  </h1>
  <h1>
    23
  </h1>
  <button
    data-test="decrease"
  >
    decrement
  </button>
  <button
    data-test="increase"
  >
    increase
  </button>
</div>
`;

exports[`Counter 2`] = `
<div>
  <h1>
    Dan_24
  </h1>
  <h1>
    24
  </h1>
  <button
    data-test="decrease"
  >
    decrement
  </button>
  <button
    data-test="increase"
  >
    increase
  </button>
</div>
`;
